programs/examples/connected/src/lib.rs

```diff
@@ -48,11 +48,20 @@ pub mod connected {
         let message = String::from_utf8(data).map_err(|_| ErrorCode::InvalidDataFormat)?;
         pda.last_message = message;
 
-        // Transfer some portion of lamports transferred from gateway to another account
-        pda.sub_lamports(amount / 2)?;
-        ctx.accounts.random_wallet.add_lamports(amount / 2)?;
+        // Split the half equally among all remaining_accounts and pda
+        let half = amount / 2;
+        let rem_accounts_len = ctx.remaining_accounts.len() as u64;
+        let share = half / rem_accounts_len;
```

Comment:
nit. would be good to move these two lines down inside the `if rem_accounts_len > 0 { // here}`?